/*
 * Copyright (c) 2008, 2011, Oracle and/or its affiliates. All rights reserved.
 *
 * Permission is hereby granted, free of charge, to any person obtaining a
 * copy of this software and associated documentation files (the "Software"),
 * to deal in the Software without restriction, including without limitation
 * the rights to use, copy, modify, merge, publish, distribute, sublicense,
 * and/or sell copies of the Software, and to permit persons to whom the
 * Software is furnished to do so, subject to the following conditions:
 *
 * The above copyright notice and this permission notice (including the next
 * paragraph) shall be included in all copies or substantial portions of the
 * Software.
 *
 * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
 * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
 * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.  IN NO EVENT SHALL
 * THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
 * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
 * FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
 * DEALINGS IN THE SOFTWARE.
 */

Fix for: 6585644 xscreensaver can cause KeyPress event loop between itself and xscreensaver-lock.
---
 driver/timers.c |    7 +++++++
 1 files changed, 7 insertions(+), 0 deletions(-)

diff --git driver/timers.c driver/timers.c
--- driver/timers.c
+++ driver/timers.c
@@ -131,6 +131,13 @@ notice_events (saver_info *si, Window window, Bool top_p)
   unsigned int nkids;
   int screen_no;
 
+  if ((si->pw_data->got_windowid) && (window == si->passwd_dialog))
+    {
+      if (p->verbose_p)
+	fprintf (stderr, "--> notice_events() breaking out of loop!\n");
+      return;
+    }
+
   if (XtWindowToWidget (si->dpy, window))
     /* If it's one of ours, don't mess up its event mask. */
     return;

